### PR TITLE
WINDUPRULE-1010 Cache Libraries override

### PR DESCRIPTION
**Pull request by PhilipCattanach**
Hi @mrizzi 
I know how busy you are so please consider this a low priority item.
I have had a try updating the rules that identify the caching technologies so that the cloud-readiness versions override vanilla technology usage versions. 
However I am having trouble with the tests for the cloud-readiness ruleset.
They work perfectly when built into the cli and executed with the cloud-readiness target.

### CODE REVIEW

File: rules/rules-reviewed/cloud-readiness/tests/embedded-cache-libraries-cloud-readiness.windup.test.xml [hunk 1]
```diff
@@ -1,11 +1,12 @@
 <?xml version="1.0"?>
-<ruletest id="embedded-cache-libraries-test" xmlns="http://windup.jboss.org/schema/jboss-ruleset" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+<ruletest id="embedded-cache-libraries-cloud-readiness-test" xmlns="http://windup.jboss.org/schema/jboss-ruleset" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
     xsi:schemaLocation="http://windup.jboss.org/schema/jboss-ruleset http://windup.jboss.org/schema/jboss-ruleset/windup-jboss-ruleset.xsd">
-    <testDataPath>data/embedded-cache-libraries</testDataPath>
+    <testDataPath>../../technology-usage/tests/data/embedded-cache-libraries</testDataPath>
```
Comment (mrizzi) [inline rules/rules-reviewed/cloud-readiness/tests/embedded-cache-libraries-cloud-readiness.windup.test.xml:4]: Also the overridden ruleset must be referenced in order for the oveeriding one to work properly
```suggestion
    <testDataPath>../../technology-usage/tests/data/embedded-cache-libraries</testDataPath>
    <rulePath>../../technology-usage/embedded-cache-libraries.windup.xml</rulePath>
```

File: rules/rules-reviewed/cloud-readiness/embedded-cache-libraries.windup.xml [hunk 2]
```diff
@@ -289,13 +289,34 @@
                 </or>
             </when>
             <perform>
-                <classification title="Caching - Spring Boot Cache library" category-id="information" effort="0">
+                <classification title="Caching - Spring Boot Cache library" category-id="cloud-mandatory" effort="5">
```
Comment (mrizzi) [inline rules/rules-reviewed/cloud-readiness/embedded-cache-libraries.windup.xml:292]: Isn't worth having this to become a `hint` considering the content it provides?
It looks more than a "simple" classification but rather a suggestion on how to migrate the embedded cache to an external caching service.

**Review by mrizzi — COMMENTED** [review]
fixed the broken test and suggested an enhancement to the overriding rules

File: rules/rules-reviewed/cloud-readiness/embedded-cache-libraries.windup.xml (hunk 2 above)
Comment (PhilipCattanach) [reply rules/rules-reviewed/cloud-readiness/embedded-cache-libraries.windup.xml:292]: @mrizzi - What is the benefit of  a hint over a classification? These issues don't have any supporting links. And they appear on the Issue report fine.

Comment (mrizzi) [reply rules/rules-reviewed/cloud-readiness/embedded-cache-libraries.windup.xml:292]: Ah, not a practical benefit on the reports but rather more a formal change.

**Comment by PhilipCattanach** [comment]
Having problems pushing to this PR so I'm going create a new one

**Pull request closed without merging**